fuel/converters/caltech101.py

```diff
@@ -0,0 +1,196 @@
+import os
+import h5py
+
+import numpy
+import scipy.misc
+
+from six.moves import range
+
+from fuel.converters.base import fill_hdf5_file, MissingInputFiles
+
+CATEGORIES = (
+    'Leopards',
+    'emu',
+    'hedgehog',
+    'binocular',
+    'cougar_body',
+    'buddha',
+    'Faces_easy',
+    'beaver',
+    'windsor_chair',
+    'yin_yang',
+    'anchor',
+    'pagoda',
+    'mayfly',
+    'flamingo_head',
+    'headphone',
+    'joshua_tree',
+    'wrench',
+    'platypus',
+    'dollar_bill',
+    'dalmatian',
+    'mandolin',
+    'llama',
+    'electric_guitar',
+    'panda',
+    'lamp',
+    'pyramid',
+    'kangaroo',
+    'strawberry',
+    'stop_sign',
+    'flamingo',
+    'gerenuk',
+    'crayfish',
+    'ketch',
+    'crocodile_head',
+    'chandelier',
+    'cellphone',
+    'brain',
+    'car_side',
+    'ferry',
+    'nautilus',
+    'BACKGROUND_Google',
+    'metronome',
+    'water_lilly',
+    'dolphin',
+    'euphonium',
+    'crocodile',
+    'Faces',
+    'sunflower',
+    'garfield',
+    'soccer_ball',
+    'stapler',
+    'scorpion',
+    'wheelchair',
+    'saxophone',
+    'starfish',
+    'lotus',
+    'okapi',
+    'octopus',
+    'hawksbill',
+    'chair',
+    'crab',
+    'menorah',
+    'helicopter',
+    'accordion',
+    'rhino',
+    'ant',
+    'bass',
+    'bonsai',
+    'butterfly',
+    'ewer',
+    'pizza',
+    'umbrella',
+    'revolver',
+    'airplanes',
+    'grand_piano',
+    'trilobite',
+    'cannon',
+    'wild_cat',
+    'inline_skate',
+    'watch',
+    'pigeon',
+    'rooster',
+    'cup',
+    'dragonfly',
+    'barrel',
+    'ceiling_fan',
+    'lobster',
+    'minaret',
+    'schooner',
+    'cougar_face',
+    'Motorbikes',
+    'laptop',
+    'elephant',
+    'sea_horse',
+    'snoopy',
+    'brontosaurus',
+    'gramophone',
+    'camera',
+    'stegosaurus',
+    'tick',
+    'scissors',
+    'ibis')
+
+NUM_TRAIN = 20
+NUM_TEST = 10
+
+
+def read_image(imfile):
```

Comment:
I'm not very fond of resizing the images to 256x256, as it hard-codes part of the pre-processing right into the converter. I think we should leave them as variable-length data and instead make the resizing part of the dataset's default transformer.